firefox-ios/firefox-ios-tests/Tests/ClientTests/Frontend/Homepage Rebuild/Mock/MockContileProvider.swift

```diff
@@ -0,0 +1,53 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Shared
+
+@testable import Client
+
+class MockContileProvider: ContileProviderInterface {
```

Comment:
copied most of this from `ContileProviderMock` but added a couple fields such as the error + empty success data so I can test the different code paths. Instead of modifying existing, I decide to create a new one and nest it under homepage rebuild tests.